### PR TITLE
implement request_id on mostro messages

### PR DESCRIPTION
**Pull request by arkanoider**
@grunch @bilthon 

not the best implementation, in case we want to do a cleaner job i think and use it in mostrod in main trunk a proper sync variable like an `Atomic` Rust is the right way.

Basically I get the `request_id` when message comes in and propagate down to the code, a bit spaghetti now, but I should have cover all the request. I used in some case to simplify my life an option ( scheduler or main resubscribe invoice cases ).

Take a look!
Will try to test now or tomorrow...

## Release Notes

- **New Features**
	- Enhanced request tracking by introducing a `request_id` parameter across multiple functions, improving message handling and traceability.
	- Updated the invoice subscription process to include a `request_id`.

- **Bug Fixes**
	- Improved error handling in various functions to ensure messages are sent with the correct request context.

- **Documentation**
	- Updated test cases to reflect changes in message structure, including the new `request_id` field.

These changes collectively enhance the clarity and management of requests within the application, ensuring a more robust user experience.

### CODE REVIEW

**Comment by arkanoider** [comment]
@coderabbitai review

File: src/app/add_invoice.rs [hunk 1]
```diff
@@ -73,8 +74,14 @@
...
-                    send_new_order_msg(Some(order.id), Action::IncorrectInvoiceAmount, None, &event.sender)
-                        .await;
+                    send_new_order_msg(
+                        msg.get_inner_message_kind().request_id,
+                        Some(order.id),
+                        Action::IncorrectInvoiceAmount,
+                        None,
+                        &event.sender,
+                    )
+                    .await;
```
[the remark this answers is not in the document]
Comment (grunch) [reply src/app/add_invoice.rs:84]: Hi @arkanoider if you made this change please solve this conversation

File: src/app/add_invoice.rs [hunk 2]
```diff
@@ -91,11 +98,19 @@
...
             order.clone().update(pool).await?;
-            send_new_order_msg(Some(order.id), Action::InvoiceUpdated, None, &buyer_pubkey).await;
+            send_new_order_msg(
+                msg.get_inner_message_kind().request_id,
+                Some(order.id),
+                Action::InvoiceUpdated,
+                None,
+                &buyer_pubkey,
+            )
+            .await;
```
[the remark this answers is not in the document]
Comment (grunch) [reply src/app/add_invoice.rs:108]: Hi @arkanoider if you made this change please solve this conversation

File: src/app/add_invoice.rs [hunk 3]
```diff
@@ -148,14 +164,23 @@ pub async fn add_invoice_action(
...
-        show_hold_invoice(my_keys, None, &buyer_pubkey, &seller_pubkey, order).await?;
+        show_hold_invoice(
+            my_keys,
+            None,
+            &buyer_pubkey,
+            &seller_pubkey,
+            order,
+            msg.get_inner_message_kind().request_id,
+        )
+        .await?;
```
[the remark this answers is not in the document]
Comment (grunch) [reply src/app/add_invoice.rs:183]: same that above

**Comment by arkanoider** [comment]
Yes! @grunch I will look tonite, I will read back rabbit suggestion and in case will remove some `request_id`, i think I overused in some invoices management function, will try to keep a cleaner situation.

File: src/app/release.rs [hunk 4]
```diff
@@ -44,7 +44,14 @@
...
 
-    send_new_order_msg(Some(order.id), Action::PaymentFailed, None, &buyer_pubkey).await;
+    send_new_order_msg(
+        request_id.unwrap_or(1),
+        Some(order.id),
+        Action::PaymentFailed,
+        None,
+        &buyer_pubkey,
+    )
+    .await;
```
[the remark this answers is not in the document]
Comment (arkanoider) [reply src/app/release.rs:54]: Now that `request_id` is an `Option` we can safely just pass it.

File: src/app/cancel.rs [hunk 5]
```diff
@@ -184,8 +204,22 @@
...
+        )
+        .await;
+        send_new_order_msg(
+            request_id,
+            Some(order.id),
+            Action::Canceled,
+            None,
+            &seller_pubkey,
+        )
+        .await;
```
[the remark this answers is not in the document]
Comment (arkanoider) [reply src/app/cancel.rs:222]: Not introducing this, imo is not so useful.

File: src/app.rs [hunk 6]
```diff
@@ -79,39 +79,49 @@ pub async fn run(
                     match message {
                         Ok(msg) => {
                             if msg.get_inner_message_kind().verify() {
+                                let request_id = msg.get_inner_message_kind().request_id;
+
```
[the remark this answers is not in the document]
Comment (arkanoider) [reply src/app.rs:83]: Now `request_id` is an `Option` we should be safe.

File: src/util.rs [hunk 7]
```diff
@@ -440,7 +445,7 @@ pub async fn invoice_subscribe(hash: Vec<u8>) -> anyhow::Result<()> {
                 let hash = bytes_to_string(msg.hash.as_ref());
                 // If this invoice was paid by the seller
                 if msg.state == InvoiceState::Accepted {
-                    if let Err(e) = flow::hold_invoice_paid(&hash).await {
+                    if let Err(e) = flow::hold_invoice_paid(&hash, request_id).await {
                         info!("Invoice flow error {e}");
                     } else {
                         info!("Invoice with hash {hash} accepted!");
```
[the remark this answers is not in the document]
Comment (arkanoider) [reply src/util.rs:451]: I think this is an overshoot, we are spawning a thread for every invoice. A locking mechanism seems not useful here.
@grunch @bilthon

**Pull request merged**